src/main/java/de/tum/in/www1/artemis/versioning/VersionRangeComparator.java

```diff
@@ -0,0 +1,127 @@
+package de.tum.in.www1.artemis.versioning;
+
+import static de.tum.in.www1.artemis.versioning.VersionRangeComparisonType.*;
+import static de.tum.in.www1.artemis.versioning.VersionRangeService.versionRangeToIntegerList;
+
+import java.util.List;
+
+import de.tum.in.www1.artemis.exception.ApiVersionRangeNotValidException;
+
+/**
+ * Holds comparison logic for {@link VersionRange} annotations.
+ */
+public class VersionRangeComparator {
+
+    /**
+     * @param a Item containing two versions sorted ASC
+     * @param b Item to be compared to containing two versions sorted ASC
+     * @return A {@link VersionRangeComparisonType}
+     */
+    public static VersionRangeComparisonType compare(VersionRange a, VersionRange b) {
+        var versions1 = versionRangeToIntegerList(a);
+        var versions2 = versionRangeToIntegerList(b);
+        if (versions1.size() == 1 && versions2.size() == 1) {
+            return compareTwoLimits(versions1, versions2);
+        }
+        else if (versions1.size() == 2 && versions2.size() == 2) {
+            return compareTwoRanges(versions1, versions2);
+        }
+        else if (versions1.size() == 1 && versions2.size() == 2) {
+            return compareLimitAndRange(versions1.getFirst(), versions2);
+        }
+        else if (versions1.size() == 2 && versions2.size() == 1) {
+            return compareLimitAndRange(versions2.getFirst(), versions1).invert();
+        }
+        throw new ApiVersionRangeNotValidException();
+    }
+
+    private static VersionRangeComparisonType compareTwoLimits(List<Integer> limit1, List<Integer> limit2) {
+        var result = Integer.compare(limit1.getFirst(), limit2.getFirst());
+        // This could be simplified but for better readability (Constants) it is not
+        if (result < 0) {
+            // First limit starts after second one. Since both are infinite second one includes first one
+            return A_INCLUDES_B;
+        }
+        else if (result > 0) {
+            // First limit starts before second one. Since both are infinite first one includes second one
+            return B_INCLUDES_A;
+        }
+        else {
+            // Both start the same point
+            return EQUALS;
+        }
+    }
+
+    private static VersionRangeComparisonType compareLimitAndRange(int limitVersion, List<Integer> range) {
+        var result = Integer.compare(limitVersion, range.get(0));
+        if (result < 1) {
```

Comment:
As far as I can tell, it is not specified what `result < 1` is supposed to mean for Integer.compare, see: https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Integer.html#compare(int,int)

Reply:
Not sure what you mean by that.
`the value 0 if x == y; a value less than 0 if x < y; and a value greater than 0 if x > y`

=> result < 1 <=> x "<=" y

Reply:
Shouldn't it be
```suggestion
        if (result < 0) {
```

Reply:
No, see the comment in the body: "Range starts with or after the limit starts". In other words, if the Limit, which is A, starts afterwards, it doesn't include the range. If it starts with the range or before it includes the range. Hence, < 0 and eq 0 are relevant here.

Reply:
But you don't check for < 0

Reply:
Got it